### PR TITLE
Initial Token Contracts

### PR DESCRIPTION
**Pull request by ejfitzgerald**
- Rework of non-primitive serialisation format
- Removal of `vm::Address::AsString()`
- Rework of the contract initialiser code
- Added unit test for the basic token contract

### CODE REVIEW

**Comment by ejfitzgerald** [comment]
FYI: @devjsc

File: libs/ledger/src/chaincode/smart_contract.cpp [hunk 1]
```diff
@@ -425,43 +419,44 @@ Contract::Status SmartContract::InvokeAction(std::string const &name, Transactio
...
   // lookup the function / entry point which will be executed
-  Script::Function const *target_function = script_->FindFunction(name);
+  Script::Function const *target_function = script_->FindFunction(init_fn_name_);
   if (target_function->num_parameters == 1)
   {
-    FETCH_LOG_INFO(LOGGING_NAME, "One argument for init - defaulting to address population");
+    FETCH_LOG_DEBUG(LOGGING_NAME, "One argument for init - defaulting to address population");
+
+    // create the public key from the identity
+    auto const identity = owner.identifier();
```
Comment (pbukva) [inline libs/ledger/src/chaincode/smart_contract.cpp]: `auto const &identity`

File: libs/ledger/src/chaincode/smart_contract_manager.cpp [hunk 2]
```diff
@@ -149,8 +149,7 @@ Contract::Status SmartContractManager::OnCreate(Transaction const &tx)
     smart_contract.Attach(state());
 
     // Dispatch to the init. method
-    auto status = smart_contract.DispatchTransaction(on_init_function, tx);
-
+    auto const status = smart_contract.DispatchInitialise(tx.signatures().begin()->first);
```
Comment (pbukva) [inline libs/ledger/src/chaincode/smart_contract_manager.cpp:152]: Here we assume some that either there will be just single singatory, or the first signatory is the one we use as identifier for the contract.

**Review by pbukva — APPROVED** [review]
LGTM

**Pull request merged**